classes/endpoint.php

```diff
@@ -45,16 +45,16 @@ public function register_routes() {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_oembed_response( WP_REST_Request $request ) {
+		if ( 'xml' === $request['format'] ) {
+			add_filter( 'rest_pre_serve_request', array( $this, 'rest_pre_serve_request' ), 10, 4 );
+		}
```

Comment:
Generally a bad idea to do this inside your callback, since the callbacks might be called multiple times in a request (if using embedding). You're better off hooking this in based on the global request data via the actions/filters in `serve_request` instead :)